app/database/admin/data_operator/index.ts

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {MM_TABLES} from '@constants/database';
+import {
+    BatchOperations,
+    DBInstance,
+    HandleBaseData,
+    HandleIsolatedEntityData,
+    RecordValue,
+} from '@typings/database/database';
+
+import DatabaseManager from '../database_manager';
+
+import {
+    operateAppRecord,
+    operateCustomEmojiRecord,
+    operateGlobalRecord,
+    operateRoleRecord,
+    operateServersRecord,
+    operateSystemRecord,
+    operateTermsOfServiceRecord,
+} from './operators';
+
+export enum OperationType {
+    CREATE = 'CREATE',
+    UPDATE = 'UPDATE',
+    DELETE = 'DELETE'
+}
+
+export enum IsolatedEntities {
+    APP= 'app',
+    GLOBAL = 'global',
+    SERVERS = 'servers',
+    CUSTOM_EMOJI = 'CustomEmoji',
+    ROLE = 'Role',
+    SYSTEM = 'System',
+    TERMS_OF_SERVICE = 'TermsOfService'
+}
+
+class DataOperator {
+    private defaultDatabase: DBInstance;
+    private serverDatabase: DBInstance;
+
+    /**
+     * handleIsolatedEntityData: Operator that handles Create/Update operations on the isolated entities as
+     * described by the IsolatedTables type
+     * @param {HandleIsolatedEntityData} entityData
+     * @param {OperationType} entityData.optType
+     * @param {IsolatedEntities} entityData.tableName
+     * @param {Records} entityData.values
+     * @returns {Promise<void>}
+     */
+    handleIsolatedEntityData = async ({optType, tableName, values}: HandleIsolatedEntityData): Promise<void> => {
+        let recordOperator;
+
+        switch (tableName) {
+        case IsolatedEntities.APP : {
```

Comment:
I think your prettier / auto-formatter might not be working correctly; an indent block is missing here too.

Reply:
Thanks for the review @shazm .  I'm adding those corrections in my next PR.